org.eclipse.lsp4j/src/main/xtend-gen/org/eclipse/lsp4j/SemanticTokens.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2016-2018 TypeFox and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+package org.eclipse.lsp4j;
+
+import com.google.common.annotations.Beta;
+import java.util.List;
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+import org.eclipse.lsp4j.util.Preconditions;
+import org.eclipse.xtext.xbase.lib.Pure;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * @since 3.16.0
+ */
+@Beta
+@SuppressWarnings("all")
+public class SemanticTokens {
+  /**
+   * An optional result id. If provided and clients support delta updating
+   * the client will include the result id in the next semantic token request.
+   * A server can then instead of computing all semantic tokens again simply
+   * send a delta.
+   */
+  private String resultId;
+  
+  /**
+   * The actual tokens.
+   */
+  @NonNull
+  private List<Integer> data;
```

Comment:
Sorry for pinging closed PR, but shouldn't we migrate `List<Integer>` to `int[]`? It's a huge list/array with a lot of `Integer` boxing/unboxing operations. Yes, `List` is more comfortable class with all it `Collection` methods. But is it fast enough?

@ericdallo @spoenemann WDYT?

Reply:
Yeah, I think it makes sense

Reply:
I would do some profiling before optimizing something like that. Java ArrayList and Integer boxing/unboxing is super optimized in the JVM, I don't know whether you'll notice any difference for real-world array sizes.